renku/core/models/calamus.py

```diff
@@ -146,6 +146,22 @@ def _deserialize(self, value, attr, data, **kwargs):
             raise ValueError("Invalid type for field {}: {}".format(self.name, type(value)))
 
 
+class DateTimeList(fields.DateTime):
+    """A DateTime field that might be a list when deserializing."""
+
+    def __init__(self, *args, **kwargs):
+        """Create an instance."""
+        super().__init__(*args, **kwargs)
+
+    def _deserialize(self, value, attr, data, **kwargs):
+        value = normalize_value(value)
+
+        if isinstance(value, (list, tuple, set)):
+            value = value[0] if len(value) > 0 else None
```

Comment:
this would now pick a somewhat random date (Since the list isn't sorted). We short sort the list and have an explicit flag that says if we pick the oldest or latest date. For instance, for a creation date, the oldest makes sense, but for a modification date, the latest makes sense.

Reply:
This is used for creation date atm and I am not sure if we want to use it for modification date ever. I sort the values to get the first one but leave the flag implementation when the need comes up.